front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -44,6 +44,18 @@ class PipelineController {
     pipelineDAO.getPipelinesByApplication(application)
   }
 
+  @RequestMapping(value = 'getById/{id:.+}', method = RequestMethod.GET)
```

Comment:
I believe the API should be similar to:

```
GET /pipelines/{id} 
GET /pipelines/{id}/history
DELETE /pipelines/{id} 
PUT /pipelines/{id} 

POST /pipelines
GET /pipelines
GET /pipelines?name=abc
GET /pipelines?application=xyz
```

In the previous comment I included {application} because /pipeline/{id} is overloaded to the extent of not being usable (_if it's an application, we expect a collection ... if it's an id, we expect a single result_).

We could bite bullet and version since what is being talked about here is a somewhat significant change (/v2/pipelines?) affecting multiple clients.